autorest/azure/token.go

```diff
@@ -104,6 +154,79 @@ func (spt *ServicePrincipalToken) EnsureFresh() error {
 
 // Refresh obtains a fresh token for the Service Principal.
 func (spt *ServicePrincipalToken) Refresh() error {
+	newToken, err := spt.GetOauthToken()
```

Comment:
Yes. This is just an artifcat of how I got here. I agree it should be back in Refresh()